### DISCUSSION TITLE
[FR] Allow to use a section as content for the pop-up

### DISCUSSION THREAD

**Discussion by Skaronator** [Clooos/Bubble-Card#682] [Feature requests]
I think it would be a good idea to make use of a section as a pop-up content source instead of having it nested in a vertical-stack.

This would allow us to use:

1. drag and drop elements around
2. resize them using the new layout feature
3. allow using the visibility tab on element level
4. make it much more simpler to edit

I am not sure how to handle it in the code, but it would be possible to hide the whole section by default if you just use the User condition and don't select any user.

**Comment by Clooos** [comment]
Hi, that's a sexy idea, and that should be totally feasible! 

I will check that soon!

**Reply by Skaronator** [reply]
Awesome!

**Reply by narfotic** [reply]
Any news on this? Would love this feature!

**Reply by contagon** [reply]
Sorry to bump an old thread, was wondering if there is any plans to implement this? I really believe it would make popups tons easier to use. Using nested grid/horizontal stacks in the vertical stack can be tedious and painful currently.

**Reply by Clooos** [reply]
Hi! I finally found a good approach, check my other comment 😄

https://github.com/Clooos/Bubble-Card/discussions/682#discussioncomment-16465960

**Comment by celindho** [comment]
This is a really good initiative! I was just about to suggest the same with almost identical motivation 👍

**Comment by iroQuai** [comment]
This is the reason I use popups a lot less than I'd want to!

**Reply by Clooos** [reply]
Hi! I finally found a good approach, check my other comment 😄

https://github.com/Clooos/Bubble-Card/discussions/682#discussioncomment-16465960

**Reply by iroQuai** [reply]
Cool i'll test soon!

**Reply by iroQuai** [reply]
Cool i'll test soon!

**Comment by cloudbr34k84** [comment]
Great suggestion! I redid all my dashboards to incorporate the sections and now getting into bubble for its popup cards, and this would be amazing

**Reply by Clooos** [reply]
Hi! I finally found a good approach, check my other comment 😄

https://github.com/Clooos/Bubble-Card/discussions/682#discussioncomment-16465960

**Comment by celindho** [comment]
Just to understand if this is feasible: Did you @Clooos happen to check if this is something that can be implemented?

**Reply by Clooos** [reply]
Since my pop-ups use another container to works, it should be feasible this way as well.

I just don't took the time to look at this yet.

**Reply by Clooos** [reply]
Hi! I finally found a good approach, check my other comment 😄

https://github.com/Clooos/Bubble-Card/discussions/682#discussioncomment-16465960

**Comment by Nemesis24** [comment]
+1

**Reply by Clooos** [reply]
Hi! I finally found a good approach, check my other comment 😄

https://github.com/Clooos/Bubble-Card/discussions/682#discussioncomment-16465960

**Comment by lmiskowicz** [comment]
Great idea!

**Reply by Clooos** [reply]
Hi! I finally found a good approach, check my other comment 😄

https://github.com/Clooos/Bubble-Card/discussions/682#discussioncomment-16465960

**Comment by contagon** [comment]
Just joined the bubble card craze and this was my first thought as well. It would be awesome to not have to deal with vertical-stacks and be able to just drag and drop everything! Would make pop-ups feel almost native to HA.

**Reply by Clooos** [reply]
Hi! I finally found a good approach, check my other comment 😄

https://github.com/Clooos/Bubble-Card/discussions/682#discussioncomment-16465960

**Comment by sunedall** [comment]
Is this still on the radar? I'm doing YAML-only.
What I'm experiencing in pop-ups is the `columns` option not taking effect because it's not in `sections`. Then I need my 50% width buttons to be inside a `horizontal-stack` in a `vertical-stack` - functionality-wise this works well enough, I just get a little itch when I compare the spacing to my cards that is not in any pop-ups 😆 

This feature would fix it, right?
**Not in pop-up:** 
<img width="524" height="130" alt="Screenshot 2026-04-02 at 22 39 31" src="https://github.com/user-attachments/assets/72e8bb94-dbb3-47b3-b7c6-85b4e4a48dee" />
**In pop-up:**
<img width="541" height="83" alt="Screenshot 2026-04-02 at 22 39 18" src="https://github.com/user-attachments/assets/8f4870fd-5555-4f29-8939-89e12009f8cd" />

**Reply by Clooos** [reply]
Hi! I looked a bit more at this possibility in the past, but this looks way too complicated. But I will work on that gap difference in the next release 🙂

**Reply by sunedall** [reply]
> Hi! I looked a bit more at this possibility in the past, but this looks way too complicated. But I will work on that gap difference in the next release 🙂

Woah! That's rapid response! Thanks a lot - and as I said, nothing is not working, so it's purely subjective preferences 😉

Thanks so much for your work, got my home up and running so smoothly with Bubble Card. Easy and accessible, huge wife approval as well 🏆

**Reply by Clooos** [reply]
You're welcome! My wife loves it as well 😄

**Reply by Clooos** [reply]
I finally found a good approach, check my other comment 😄

https://github.com/Clooos/Bubble-Card/discussions/682#discussioncomment-16465960

**Comment by Clooos** [comment]
Hi everyone! 

Here is some teasing for the upcoming v3.2.0 release, I finally found a good approach that I really love! 🍻

## Before migration (vertical-stack approach):

<img width="968" height="938" alt="image" src="https://github.com/user-attachments/assets/f737ecd2-7df3-4be0-987a-e5fe89724b9b" />  

## After migration (section view with drag and drop, no more vertical stack):

<img width="968" height="937" alt="image" src="https://github.com/user-attachments/assets/52f794d0-f1b1-405e-ae45-75f2b9383e6d" />

<br><br>

This new approach allows you to use columns and rows directly on each card, while also supporting drag and drop, duplicate, copy, cut, paste... just like in a section view since this is the exact same component! Removing the need for vertical stacks also improves performance and fixes many other issues! The preview in the editor will remain unchanged.

The migration button will automatically update your existing popups to the new system. This is not a breaking change, and current popups will continue to work. Migration is recommended, although some modules may not be supported (but most will be).

Much more to come! 🥳

**Reply by Skaronator** [reply]
Awesome! Thanks @Clooos you rock! :D
